### PR TITLE
Repro #30312 - Cannot filter directly from table column header when Question is Summarized

### LINKED ISSUE #30312: Cannot filter directly from table column header when Question is Summarized
### Describe the bug

There is no option to add a filter directly from table column header. 

image

### To Reproduce

1. Go to New Question -> Orders -> Count and Grouped by Created_at: Month

image

2. Visualize as Table -> Click on Count Header -> Filter by this Column -> Equal to 10 -> the Add Filter is Faded

image

### Expected behavior

Able to filter via the Column Header

### Logs

None pop up

### Information about your Metabase installation

```JSON
Able to replicate this on 1.45.3.1 and `master`
```

### Severity

There is a workaround in which you filter from the Editor

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by kamilmielnik**
Closes #30312

Stress test: https://github.com/metabase/metabase/actions/runs/8140158201

### CODE REVIEW

File: e2e/test/scenarios/filters/reproductions/30312-drill-filter-on-aggregated-column.cy.spec.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+const CREATED_AT_BREAKOUT = [
+  "field",
+  ORDERS.CREATED_AT,
+  {
+    "base-type": "type/DateTime",
+    "temporal-unit": "month",
+  },
+];
+
+describe("issue 30312", function () {
```
Comment (nemanjaglumac) [inline e2e/test/scenarios/filters/reproductions/30312-drill-filter-on-aggregated-column.cy.spec.js:15]:
```suggestion
describe("issue 30312", () => {
```

File: e2e/test/scenarios/filters/reproductions/30312-drill-filter-on-aggregated-column.cy.spec.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+      cy.findByText("Filter by this column").click();
+      cy.findByPlaceholderText("Enter a number").type("10");
+      cy.realPress("Tab");
+      cy.button("Add filter").should("be.enabled").click();
+    });
+
+    cy.findByTestId("qb-filters-panel").should(
+      "contain.text",
+      "Count is equal to 10",
+    );
```
Comment (nemanjaglumac) [inline e2e/test/scenarios/filters/reproductions/30312-drill-filter-on-aggregated-column.cy.spec.js:50]:
```suggestion
    cy.findByTestId("filter-pill").should(
      "have.text",
      "Count is equal to 10",
    );
```

Comment (nemanjaglumac) [reply e2e/test/scenarios/filters/reproductions/30312-drill-filter-on-aggregated-column.cy.spec.js:50]: Nit: It wouldn't hurt to add one more assertion for the actual result. For the count=10, it's "No results", so you can either do that or choose any of the existing values
image

Comment (kamilmielnik) [reply e2e/test/scenarios/filters/reproductions/30312-drill-filter-on-aggregated-column.cy.spec.js]: 2023f4fd04cbf39ef3da42b3ab1f86621a07dad1

**Review by nemanjaglumac — APPROVED** [review]

**Pull request merged**